hack/generator/pkg/jsonast/schema_abstraction_gojson.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package jsonast
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// GoJSONSchema implements the Schema abstraction for gojsonschema
+type GoJSONSchema struct {
+	schema *gojsonschema.SubSchema
+}
+
+func MakeGoJSONSchema(schema *gojsonschema.SubSchema) Schema {
+	return GoJSONSchema{schema}
+}
+
+var _ Schema = GoJSONSchema{}
+
+func transformGoJSONSlice(slice []*gojsonschema.SubSchema) []Schema {
+	result := make([]Schema, len(slice))
+	for i := range slice {
+		result[i] = GoJSONSchema{slice[i]}
+	}
+
+	return result
+}
+
+func (it GoJSONSchema) URL() *url.URL {
```

Comment:
I think idiomatic Go would suggest a more informative name for the receiver than `it` - how about `schema` ?